Docs for using a new SOAP plugin. (#356)
playframework/play-soap: docs/modules/ROOT/pages/index.adoc

= Play SOAP

Play SOAP allows an application to make calls on a remote web service using SOAP. It provides a reactive interface to doing so, making HTTP requests asynchronously and returning promises/futures of the result.

== JAX WS support

> Play SOAP builds on the JAX WS spec, but doesn't implement it exactly. JAX WS, while it does have support for making asynchronous calls, this support is somewhat clumsy, requiring all asynchronous methods to have an `Async` suffix, and requiring the passing of an `AsyncHandler` argument to handle the response, which makes it awkward to integrate into an asynchronous framework since `AsyncHandler's` do not compose well with other asynchronous constructs. This support could be described as a second class citizen, bolted on to the spec as an after thought.

Reviewer: Gramatically, the apostrophe doesn't actually belong here. I would also keep the s out of the code block, since the only portion that is code is `AsyncHandler`.  So, `AsyncHandler`s.

Reviewer: ```suggestion
Play SOAP builds on the JAX WS spec, but doesn't implement it exactly. JAX WS does have support for making asynchronous calls, but this support is somewhat clumsy. It requires all asynchronous methods to have an `Async` suffix, and requires the passing of an `AsyncHandler` argument to handle the response. This makes it awkward to integrate into an asynchronous framework, since `AsyncHandler`s do not compose well with other asynchronous constructs. This support could be described as a second class citizen, bolted on to the spec as an afterthought.
```
